File: backend/hotelEase/hotelEase.Model/Review.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace hotelEase.Model
{
    public class Review
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public int HotelId { get; set; }

        public int ReservationId { get; set; }

        public int Rating { get; set; }

        public string? Comment { get; set; }

        public DateTime? ReviewDate { get; set; }
        public bool? IsDeleted { get; set; }
        public DateTime? DeletedTime { get; set; }
    }
}

File: UI/hotelease_mobile_new/lib/models/review.dart
import 'package:json_annotation/json_annotation.dart';

part 'review.g.dart';

@JsonSerializable()
class Review {
  int? id;
  int? userId;
  int? hotelId;
  int? reservationId;
  int rating;
  String? comment;
  DateTime? reviewDate;

  Review({
    this.id,
    this.userId,
    this.hotelId,
    this.reservationId,
    required this.rating,
    this.comment,
    this.reviewDate,
  });

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);

  Map<String, dynamic> toJson() => _$ReviewToJson(this);
}
